### PR TITLE
Tyler hernandez development

### PR DESCRIPTION
**Pull request by TylerHernandez**

### CODE REVIEW

File: portfolio/src/main/webapp/percy.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8">
+    <title>Tyler's Portfolio</title>
+    <link rel="stylesheet" href="style.css">
+    <!--<script src="script.js"></script>-->
```
Comment (mutisog) [inline portfolio/src/main/webapp/percy.html:7]: Generally, best practice is to not submit commented out code.

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -26,3 +26,15 @@ function addRandomGreeting() {
...
+function addRandomFact(){
+    const randomFacts=
+        ['I live in Manhattan, New York', 'I have the world\'s sleepiest dog', 'I absolutely drank too much coffee today'];
+
+    //Pick a random fact
+    const randomFact=randomFacts[Math.floor(Math.random() * randomFacts.length)];
+
+    //Add it to the page
+    const factContainer = document.getElementById('fact-container');
+    factContainer.innerText=randomFact;
```
Comment (mutisog) [inline portfolio/src/main/webapp/script.js:26]: Ideally, the code should follow consistent styling and formatting, for easier readability. In this case, the spacing around the "=" sign is not consistent. In general, prefer more whitespace i.e. "abcd = efgh".

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -19,14 +19,35 @@
...
+    ArrayList<String> list = new ArrayList<String>();
+    list.add("Hey this is message number 1");
+    list.add("AND this is message number 2");
+    list.add("couldn't forget about message #3");
+    String json = convertToJson(list);
+    System.out.println(json);
+    response.getWriter().println(json);
+  }
+
+   private String convertToJson(ArrayList<String> list) {
```
Comment (mutisog) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Curious: what would happen if convertToJson(...) was called with an input of > 3 strings?

File: portfolio/src/main/webapp/style.css [hunk 4]
```diff
@@ -1,9 +1,59 @@
...
   margin-left: auto;
   margin-right: auto;
-  width: 650px;
+  width: 1050px;
+  background-color:lightblue;
+  text-align: center;
+  font-size:50px;
+}
+
+#title {
```
Comment (mutisog) [inline portfolio/src/main/webapp/style.css:17]: Good work on the styling!

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -38,3 +38,68 @@ function addRandomFact(){
     const factContainer = document.getElementById('fact-container');
     factContainer.innerText=randomFact;
 }
+
+function getRandomQuote() {
```
Comment (mutisog) [inline portfolio/src/main/webapp/script.js:29]: I like! The code flow in these functions were easy to follow ::thumbsup::

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (mutisog) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Curious: what would happen if convertToJson(...) was called with an input of > 3 strings?

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (mutisog) [inline portfolio/src/main/webapp/script.js:29]: I like! The code flow in these functions is easy to follow :thumbsup:

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (mutisog) [inline portfolio/src/main/webapp/script.js:26]: Ideally, the code should follow consistent styling and formatting, for easier readability. In this case, the spacing around the "=" sign is not consistent. In general, prefer more whitespace i.e. "abcd = efgh".

File: portfolio/src/main/webapp/style.css (hunk 4 above)
Comment (mutisog) [inline portfolio/src/main/webapp/style.css:17]: Good work on the styling!

File: portfolio/src/main/webapp/percy.html (hunk 1 above)
Comment (mutisog) [inline portfolio/src/main/webapp/percy.html:7]: Generally, best practice is to not submit commented out code.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -22,44 +22,53 @@
...
+    // Respond with the result.
+    response.setContentType("text/html;");
+    response.getWriter().println(text);
+    list.add(text);
+    response.sendRedirect("/index.html");
+    //System.out.println(text);
+  }
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
```
Comment (mutisog) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:83]: nit: for when you have time: read about the ternary operator.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -22,44 +22,53 @@
...
+      return defaultValue;
+    }
+    return value;
   }
-// this allows for as many elements as possible in list
+// use this if more than one element in list
   private String nconvertToJson(ArrayList<String> list) {
+      //if only one element in list
```
Comment (mutisog) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: The comments in line 53 and 55 seem contradictory... ?

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -80,26 +68,47 @@ async function getData() {
...
+        createListElement(data[key]));
         console.log("hit");
-        console.log(data);
+    }
   });
 }
 
+function layoutComments() {
+    
+}
```
Comment (mutisog) [inline portfolio/src/main/webapp/script.js]: Is this function needed?

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -12,17 +12,86 @@
...
+
+function getRandomQuote() {
+  console.log('Fetching a random quote.');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
```
Comment (albert-g3) [inline portfolio/src/main/webapp/script.js:36]: can you handle error case? e.g. 
`.catch((error) => {
  console.error('Error:', error);
});`
[full api](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)

**Pull request closed without merging**